helix-term/src/application.rs

```diff
@@ -417,6 +433,7 @@ impl Application {
     fn refresh_config(&mut self) {
```

Comment:
I think the `refresh_config`, `refresh_language_config` and `refresh_theme` commands could use some refactoring to improve the error handling. With this change, if you have an error in your language config and a theme error, you only end up seeing the theme error since it overwrites the `editor.set_error` from `refresh_language_config`

`refresh_language_config` and `refresh_theme` can return `Result<(), Error>`s instead and then use the anyhow helpers like `anyhow::anyhow!` or `anyhow::bail!` to format the error messages and return early in cases of errors. Then `refresh_config` can set the editor error if there are any failures:

```rs
fn refresh_config(&mut self) {
    let mut refresh_config = || -> Result<(), Error> {
        let config = Config::load_default()
            .map_err(|err| anyhow::anyhow!("failed to load config: {}", err))?;
        self.refresh_language_config()?;
        self.refresh_theme(&config)?;

        // Store new config
        self.config.store(Arc::new(config));

        Ok(())
    };
    if let Err(err) = refresh_config() {
        self.editor.set_error(err.to_string());
    }
}
```

And the other functions can use `?` to return errors early and remove the duplicated error handling code:

```rs
fn refresh_language_config(&mut self) -> Result<(), Error> {
    let syntax_config = helix_core::config::user_syntax_loader()?;
    self.syn_loader = // ...
    // ...
    Ok(())
}

Reply:
Liked the suggestion, code looks a lot cleaner now.

I'm new to rust so recommendations like this are always welcomed